src/lib.rs

```diff
@@ -284,10 +272,9 @@ pub(crate) unsafe fn rav1d_open(c_out: &mut *mut Rav1dContext, s: &Rav1dSettings
     addr_of_mut!((*c).cache).write(Default::default());
     addr_of_mut!((*c).refs).write(Default::default());
     for n in 0..n_fc {
-        let fc = &mut *((*c).fc).offset(n as isize);
+        let fc = &mut (*c).fc[n];
         addr_of_mut!(fc.data).write(RwLock::new(Rav1dFrameData::zeroed()));
         let f = fc.data.get_mut().unwrap();
-        f.index = n;
         addr_of_mut!(fc.in_cdf).write(RwLock::new(Default::default()));
         addr_of_mut!(fc.task_thread.tasks).write(UnsafeCell::new(Default::default()));
         addr_of_mut!(fc.task_thread.retval).write(Mutex::new(Ok(())));
```

Comment:
```suggestion
```

I can't comment on the next lines, but all of the lines up to 302/289 can be removed, all of the ones initializing `fc` since that's already done on construction now.